### PR TITLE
Update the synthetic data demo

### LINKED ISSUE #63: API bug in synthetic-data example
In the synthetic-data example a `value_server_threshold` parameter is passed to `make_queue_pairs`, however, this parameter is not part of the function signature. Should it be updated to `proxystore_threshold`? In addition, is it necessary to specify the `proxystore_name` parameter for `proxystore_threshold` to have an effect? Thanks!

Code:
https://github.com/exalearn/colmena/blob/master/demo_apps/synthetic-data/synthetic.py#L159
```
client_queues, server_queues = make_queue_pairs(
        args.redis_host,
        args.redis_port,
        topics=["generate"],
        serialization_method="pickle",
        keep_inputs=False,
        value_server_threshold=value_server_threshold,
    )
```

### PR DESCRIPTION
**Pull request by gpauloski**
I forgot to open a PR for this when I finished it at the start of summer 🤦 .

- Fixes #63 

## Major Changes
- Synthetic app now supports Parsl and FuncX backends.
- Specify proxystore backend. 

## Minor Changes
- Better argument parsing.
- Removed config file parsing support (command line args + bash scripts suffice).
- Added more performance markers to `Result` objects.

## Testing
Since the branch was out-of-date since I last tested it, I updated from master this morning and ran tests with Parsl and FuncX and with and without ProxyStore.

As a side note, there is an issue in FuncX that will cause errors with the latest FuncX version on PyPI. Specifially, https://github.com/funcx-faas/funcX/commit/9eef754429dc83b9796ce9534b0aeaafd9c4d513 introduces a bug where a property is attempted to be set without having a setter and https://github.com/funcx-faas/funcX/commit/206d1f1360ecb0134d9d7305f5423d4257dd67b1 fixes it. FuncX 0.3.5 worked as it was before the broken commit, but for versions 0.3.6-0.3.10, you may need to add the patch yourself.

### CODE REVIEW

File: colmena/task_server/parsl.py [hunk 1]
```diff
@@ -402,6 +402,7 @@ def _submit(self, task: Result, topic: str) -> Optional[Future]:
             method = task.method
 
         # Submit the application
+        task.mark_task_submitted()
```
Comment (WardLT) [inline colmena/task_server/parsl.py:405]: We should emphasize that this the time we start the submission process. If we wanted the time after it's done submitting, we could store that information in the task server and then inject it into the result object before sending it back to the client.

File: colmena/models.py [hunk 2]
```diff
@@ -134,8 +134,11 @@ class Result(BaseModel):
     time_created: float = Field(None, description="Time this value object was created")
     time_input_received: float = Field(None, description="Time the inputs was received by the task server")
     time_compute_started: float = Field(None, description="Time workflow process began executing a task")
+    time_compute_ended: float = Field(None, description="Time worflow process finished executing a task")
```
Comment (WardLT) [inline colmena/models.py:137]: Thank you for adding this. The "time_running" + "time_compute_started" are added together so often we might as well do it for them.

Could you fix the typo of worflow -> workflow?

File: colmena/task_server/base.py [hunk 3]
```diff
@@ -123,6 +123,8 @@ def perform_callback(self, future: Future, result: Result, topic: str):
             # If not, the result object is the one we need
             result = future.result()
 
+        result.mark_task_received()
+
```
Comment (WardLT) [inline colmena/task_server/base.py:127]: I agree with the spot here.

**Review by WardLT — COMMENTED** [review]
Two things: 
1. A typo
2. The "time submitted" captures the time we start rather than finish submitting the task. Should we change that? It'd require holding on to the time in the task server until the result object comes back.

**Comment by WardLT** [comment]
Regarding your FuncX note, do we need to adjust Colmena in the future?

File: colmena/task_server/parsl.py (hunk 1 above)
Comment (gpauloski) [reply colmena/task_server/parsl.py:405]: Makes sense. Would changing the name from `task_submitted` -> `start_task_submission` and updating the field description be sufficient?

Comment (WardLT) [reply colmena/task_server/parsl.py:405]: Yea, for the scope of this PR. Let's re-consider which ones we measure and how we describe them at some point

**Comment by gpauloski** [comment]
> Regarding your FuncX note, do we need to adjust Colmena in the future?

I think the most we could reasonably do is pin Colmena to `funcx<=0.3.5` for now and then update when FuncX v0.4.0 is released. I briefly tried an alpha release for FuncX which failed as well, but I did not look into it too quickly.

**Comment by WardLT** [comment]
Could you pin to <0.3.5 for this PR? Hopefully 🤞, 0.4.0 isn't far off

**Comment by gpauloski** [comment]
I addressed the prior issues and pinned to <=0.3.6 (I git blamed the wrong line so the issue was actually added in 0.3.7).

**Pull request merged**